### PR TITLE
Weekly checkout

### PR DESCRIPTION
**Pull request by pvighi**
This PR adds support for guardian weekly plans. 
Before we can merge this we have to:

- ~~Merge the [PR](/guardian/membership-common/pull/336) that adds these plans to membership commons and update this code to use this newly released version.~~

- ~~Get confirmation of the final decision regarding the channel islands~~ **channel islands are in the uk delivery zone**

- ~~Make sure Zuora is configured correctly in PROD~~

### CODE REVIEW

File: app/controllers/Checkout.scala [hunk 1]
```diff
@@ -37,6 +36,14 @@ import scalaz.{NonEmptyList, OptionT}
 
 object Checkout extends Controller with LazyLogging with CatalogProvider {
 
+  val ukAndIsleOfMan = CountryGroup.UK.copy(countries = List(Country.UK, Country("IM", "Isle of Man")))
+  val weeklyUkNorthAmericaGroups = List (ukAndIsleOfMan, CountryGroup.US, CountryGroup.Canada)
+  val weeklyRowGroups = {
+    val rowUk = CountryGroup("Row Uk", "uk", None, CountryGroup.UK.countries.filterNot(ukAndIsleOfMan.countries.contains(_)), GBP, PostCode)
+    rowUk :: CountryGroup.allGroups.filterNot(group => (CountryGroup.UK :: weeklyUkNorthAmericaGroups) contains group)
+  }
```
Comment (pvighi) [inline app/controllers/Checkout.scala]: Are we able to deliver to all countries of all countrygroups or do we need a specific list of countries for Guardian Weekly ROW delivery?
This list currently include things like Antarctica

File: app/controllers/Checkout.scala [hunk 2]
```diff
@@ -37,6 +36,22 @@ import scalaz.{NonEmptyList, OptionT}
...
+    countries = List(
+      Country.UK,
+      Country("GG", "Guernsey"),
+      Country("IM", "Isle of Man"),
+      Country("JE", "Jersey")
+    ))
+
+  val weeklyUkNorthAmericaGroups = List (weeklyUkCountries, CountryGroup.US, CountryGroup.Canada)
+  val weeklyRowGroups = {
+    val rowUk = CountryGroup("Row Uk", "uk", None, CountryGroup.UK.countries.filterNot(weeklyUkCountries.countries.contains(_)), GBP, PostCode)
```
Comment (paulbrown1982) [inline app/controllers/Checkout.scala]: I think the default currency for ROW countries is USD not GBP

Comment (pvighi) [inline app/controllers/Checkout.scala]: this "rowUk" is just an intermediate country group I defined for the Countries in the Uk CountryGroup that are not included in the Uk delivery zone so they use pounds

File: app/views/checkout/payment.scala.html [hunk 3]
```diff
@@ -154,24 +159,26 @@
...
+                            @defining(if(productData.isDigitalPack) "#billingAddress" else "#deliveryDetails") { goto =>
+                                @continueButton(goto, "js-checkout-your-details-submit")
+                            }
                         </div>
                     </fieldset>
                 </div>
 
                 @* ===== Delivery details ===== *@
-                @if(productData.isHomeDelivery || productData.isVoucher) {
+                @if(productData.isPhysical) {
```
Comment (paulbrown1982) [inline app/views/checkout/payment.scala.html]: 👍

File: app/views/fragments/checkout/fieldsBilling.scala.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
 @import views.support.PlanOps._
 @import com.gu.memsub.subsv2.CatalogPlan
 
-@(billingAddress: Option[Address], canBeHidden: Boolean, countriesWithCurrency: List[CountryWithCurrency], plan: CatalogPlan.Paid)
+@(billingAddress: Option[Address], canBeHidden: Boolean, countriesWithCurrency: List[CountryWithCurrency], plan: CatalogPlan.Paid, isCurrencyAddress : Boolean)
```
Comment (paulbrown1982) [inline app/views/fragments/checkout/fieldsBilling.scala.html]: maybe rename isCurrencyAddress to: determinesCurrency

File: assets/javascripts/modules/checkout/fieldSwitcher.js [hunk 5]
```diff
@@ -56,28 +91,23 @@ define([
             redrawAddressField($subdivision, newSubdivision, model.subdivision);
         };
 
-        // Change the payment method to Direct Debit for UK users,
-        // Credit Card for international users
-        var selectPaymentMethod = function (country) {
-            if (country === 'GB') {
-                check(formElements.$DIRECT_DEBIT_TYPE[0]);
+        var redrawCountryOverride = function (selectedCurrency) {
+            $('#currency-override-checkbox').attr('checked', false);
```
Comment (paulbrown1982) [inline assets/javascripts/modules/checkout/fieldSwitcher.js]: See comment above about formElements.js

**Comment by pvighi** [comment]
@johnduffell , @jacobwinch any comments?

**Comment by johnduffell** [comment]
looks ok from a quick look through, there's a lot there, anywhere in particular you're worried about? any tests needing added?
👍

**Pull request merged**